packages/field-base/test/input-mixin.test.js

```diff
@@ -0,0 +1,95 @@
+import { expect } from '@esm-bundle/chai';
+import { fixtureSync } from '@vaadin/testing-helpers';
+import { PolymerElement, html } from '@polymer/polymer/polymer-element.js';
+import { InputMixin } from '../src/input-mixin.js';
+
+customElements.define(
+  'input-mixin-element',
+  class extends InputMixin(PolymerElement) {
+    static get template() {
+      return html`<slot name="input"></slot>`;
+    }
+  }
+);
+
+customElements.define(
+  'input-mixin-number-element',
+  class extends InputMixin(PolymerElement) {
+    static get template() {
+      return html`<slot name="input"></slot>`;
+    }
+
+    ready() {
+      super.ready();
+
+      // Set readOnly property
+      this._setType('number');
+    }
+  }
+);
+
+describe('input-mixin', () => {
+  let element, input;
+
+  describe('default', () => {
+    beforeEach(() => {
+      element = fixtureSync(`<input-mixin-element></input-mixin-element>`);
+      input = element.querySelector('[slot=input]');
+    });
+
+    it('should create an input element', () => {
+      expect(input instanceof HTMLInputElement).to.be.true;
+    });
+
+    it('should store a reference to the input', () => {
+      expect(element._inputNode).to.equal(input);
+    });
+
+    it('should have a read-only type property', () => {
+      expect(element.type).to.be.not.ok;
+      element.type = 'number';
+      expect(element.type).to.be.not.ok;
+    });
+
+    it('should have an empty name by default', () => {
+      expect(input.name).to.be.not.ok;
```

Comment:
> expect(0).to.equal(0); // Recommended
> expect(0).to.not.be.ok; // Not recommended

https://www.chaijs.com/api/bdd/#method_ok

Note that using `.to.be.not.ok` causes the test to successfully pass if `input.name === 0`, which is probably not expected behavior.

Reply:
Fixed.